src/utils/extractTestResults.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import { DynamoDBRecord } from 'aws-lambda';
+import { DynamoDB } from 'aws-sdk';
+import { DateTime } from 'luxon';
+import { TestResult } from './testResult';
+import { MCRequest } from './MCRequest';
+import logger from '../observability/logger';
+
+/**
+ * This is used to extract the relevant fields from the test record that is
+ * required to be sent to MC in order to  clear prohibitions
+ * @param record
+ */
+export const extractMCTestResults = (record: DynamoDBRecord): MCRequest[] => {
+  try {
+    const data = DynamoDB.Converter.unmarshall(record.dynamodb.NewImage);
+    const mcRequest: MCRequest[] = data.testTypes
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      .filter((x) => x.testTypeName.toLowerCase().includes('prohibition clearance'))
+      .filter((x) => (x.testResult === ('pass') || x.testResult === ('prs')))
+      .filter(() => (data.vehicleType === 'hgv') || data.vehicleType === 'psv' || data.vehicleType === 'trl')
+      .filter(() => data.testStatus === 'submitted')
+      .map((x: TestResult) => ({
+        vehicleIdentifier: data.vrm,
+        testDate: isoDateFormatter(x.testTypeEndTimestamp),
+        vin: data.vin,
+        testResult: calculateTestResult(x),
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+        hgvPsvTrailFlag: calculateTrailFlag(data.vehicleType),
```

Comment:
I'm guessing that your input for this function will always be `hgv`, `psv` or `trl` based on the filtering you're doing at the top. I'm wondering whether you need the `calculateTrailFlag` function and instead can just use:

`hsvPsvTrailFlag: data.vehicleType.charAt(0).toUpperCase()`